daemon/cmd/daemon_main.go

```diff
@@ -1748,6 +1747,16 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 	bootstrapStats.enableConntrack.Start()
 	log.Info("Starting connection tracking garbage collector")
 	params.CTNATMapGC.Enable()
+	params.CTNATMapGC.Observe4().Observe(d.ctx, ctmap.NatMapNext4, func(err error) {
+		if err != nil {
+			log.WithError(err).Error("CT map GC for ipv4 unexpectedly stopped. The stale NAT entries will not be purged anymore.")
```

Comment:
Is this true, an error will prevent future GC? This doesn't seem like an acceptable behaviour for production code.

Reply:
This gets called when the observable is closed, which should never happen.   I think it could be argued that we may want either a Fatal or a "BUG: ..." type message here.

Reply:
Does the observable get closed on shutdown?

Reply:
> Does the observable get closed on shutdown?

Yes, we call `complete(nil)` and close it.
https://github.com/cilium/cilium/pull/36084/files#diff-980bde7f9870d20814a66e921c1e040f4b7e5f196bad466ea04afe0077cb8562R101-R102

Reply:
> Though if there's any chance we're wrong it's probably better to log an error than to fatal out.

This code logs an error if something wrong happens.

Reply:
If this error handling seems to cause confusion, should we remove it?

Reply:
@cilium/sig-foundations what's best practice here? It looks to me like the implementation in this PR only ever calls `complete(nil)`, so there's no error to check. Given that this can never fail, what should callers provide for the `complete()` function?

Reply:
I guess the simple answer would be `Observe(...., func(error){})`

Reply:
@pippolo84 @tommyp1ckles WDYT? Shall we remove this error handling?

Reply:
`func(error){}` is fine

Reply:
Removed the error handling